src/main/java/org/broadinstitute/hellbender/utils/runtime/StreamingProcessController.java

```diff
@@ -0,0 +1,598 @@
+                        commandString,
+                        journalingFile.getAbsolutePath()), e);
+            }
+            System.err.println(String.format("Enabling streaming process journaling file %s", journalingFileName));
+        }
+
+        /**
+         * Record outbound data being written to a remote process.
+         * @param line line being written
+         */
+        public void writeOutbound(final String line) {
+            try {
+                if (journalingFileWriter != null) {
+                    journalingFileWriter.write("Sending: [");
+                    journalingFileWriter.write(line);
+                    journalingFileWriter.write("]\n\n");
+                    journalingFileWriter.flush();
+                }
+            } catch (IOException e) {
+                throw new GATKException("Error writing to output to process journal", e);
+            }
+        }
+
+        /**
+         * Record inbound data being received from a remote process.
+         * @param stdout Data receieved from stdout.
+         * @param stderr Data receieved from stderr.
+         */
+        public void writeInbound(final StreamOutput stdout, final StreamOutput stderr) {
+
+            if (journalingFileWriter != null) {
+                try {
+                    if (stdout != null) {
+                        journalingFileWriter.write("Received from stdout: [");
+                        journalingFileWriter.write(stdout.getBufferString());
+                        journalingFileWriter.write("]\n");
+                    }
+                    if (stderr != null) {
+                        journalingFileWriter.write("Received from stderr: [");
+                        journalingFileWriter.write(stderr.getBufferString());
+                        journalingFileWriter.write("]\n");
+                    }
+                    journalingFileWriter.write("\n");
+                    journalingFileWriter.flush();
+                } catch (IOException e) {
+                    throw new GATKException(String.format("Error writing to journaling file %s", journalingFile.getAbsolutePath()), e);
+                }
+            }
+        }
+
+        /**
+         * Write an advisory log message to the journal.
+         * @param message Message to be written to the journal.
+         */
+        public void writeLogMessage(final String message) {
+            if (journalingFileWriter != null) {
+                try {
+                    journalingFileWriter.write(message);
+                    journalingFileWriter.flush();
+                } catch (IOException e) {
+                    throw new GATKException(String.format("Error writing to journaling file %s", journalingFile.getAbsolutePath()), e);
+                }
+            }
+        }
+
+        /**
+         * Close the journal file writer.
+         */
+        public void close() {
+            try {
+                if (journalingFileWriter != null) {
```

Comment:
I might log a message here that the journal was closed so you can tell if it ended due to an expected closing or if it crashed.

Reply:
Makes sense. Done.